server/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -63,7 +72,12 @@
     private static final ParseField SLOP_FIELD = new ParseField("slop");
     private static final ParseField ZERO_TERMS_QUERY_FIELD = new ParseField("zero_terms_query");
     private static final ParseField LENIENT_FIELD = new ParseField("lenient");
-    private static final ParseField CUTOFF_FREQUENCY_FIELD = new ParseField("cutoff_frequency");
+    /**
+     * @deprecated Since max_score optimization landed in 7.0, normal MultiMatchQuery
+     *             will achieve the same result without any configuration.
+     */
+    @Deprecated
```

Comment:
This is not needed, the deprecated annotation is for java user that uses MultiMatchQueryBuilder#cutoffFrequency in their code.